File: Day1/check_prime/check_prime.cpp
// To accept a number and display whether the number is prime or not

#include <iostream>
using namespace std;

bool isprime(int n) {
   // takes an integer input and checks whether itss prime or not
   if (n <= 1)
      return false;
   for (int i = 2; i*i <= n; i++) { 
      // sq.root of n is most optimal because if n = a*b then at least one of those has to be less than root n
      if (n%i == 0)
         return false;
   }
   return true;  
}

int main() {
   int number;
   cout << "Enter a Number : ";
   cin >> number;
   if (isprime(number))
      cout << "is a prime" << endl;
   else
      cout << "is not a prime" << endl;
   return 0;
}

File: Day1/check_prime/check_prime.java
import java.util.Scanner;

public class check_prime {
    public static void main(String[] args) {
        System.out.println("Enter a Number");
        try (Scanner sc = new Scanner(System.in)) {
            int num = sc.nextInt();
            boolean isPrime = isPrime(num);

            if (isPrime)
                System.out.println(num + " is Prime");
            else
                System.out.println(num + " is not Prime");
        }
    }

    public static boolean isPrime(int num) {
        if (num <= 1)
            return false;
        if (num == 2)
            return true;
        if (num%2 == 0)
            return false;
        for (int i = 3; i <= Math.sqrt(num); i += 2) {
            if (num%i == 0)
                return false;
        }
        return true;
    }
}
